### PR TITLE
fix gptj training issue

### PR DESCRIPTION
**Pull request by BaihuiJin**
# What does this PR do?

Fix training runtime error caused by lack of sin and cos value missing when load from checkpoint.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/habana/transformers/models/gptj/modeling_gptj.py [hunk 1]
```diff
@@ -390,7 +390,7 @@ def gaudi_gptj_model_forward(
             def create_custom_forward(module):
                 def custom_forward(*inputs):
                     # None for past_key_value
-                    return module(*inputs, use_cache, output_attentions)
+                    return module(*inputs, use_cache, output_attentions, token_idx, sin, cos)
```
Comment (regisss) [inline optimum/habana/transformers/models/gptj/modeling_gptj.py:393]: `token_idx` should be False when using gradient checkpointing right?

Comment (BaihuiJin) [reply optimum/habana/transformers/models/gptj/modeling_gptj.py:393]: It's `None` actually

Comment (BaihuiJin) [reply optimum/habana/transformers/models/gptj/modeling_gptj.py]: fixed to `None`

**Review by regisss — APPROVED** [review]
LGTM

**Pull request merged**